mood-analysis.rb

```diff
@@ -28,5 +35,49 @@ def analyze_mood(words)
   "05/11 Yay, yay, yay! I'm having a awfuly great day."
 ]
 
-puts analyze_mood(text[0])
-puts analyze_mood(text[1])
+
+  text.each do |entry|
+  puts "#{entry[0..4]} #{analyze_mood(entry)}"
+  end
+
+  #determine how many logged entries it takes until there have been three :-) happy days.
+  def happy_days(text)
+    good_days = 0
+    day_count = 0
+    text.each do |text|
```

Comment:
It appears that ruby is smart enough to figure out that you named both the array and each entry in the array `text`, but this is not great convention.